crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -88,6 +102,66 @@ impl PubGrubPriorities {
             PubGrubPackageInner::Package { name, .. } => self.0.get(name).copied(),
         }
     }
+
+    /// Returns whether the priority was changed, i.e., it's the first time we hit this condition
+    /// for the package.
+    pub(crate) fn make_conflict_early(&mut self, package: &PubGrubPackage) -> bool {
+        let next = self.0.len();
+        let Some(name) = package.name_no_root() else {
+            // Not a correctness bug
+            if cfg!(debug_assertions) {
+                panic!("URL packages must not be involved in conflict handling")
+            } else {
+                return false;
+            }
+        };
+        match self.0.entry(name.clone()) {
+            std::collections::hash_map::Entry::Occupied(mut entry) => {
+                if matches!(entry.get(), PubGrubPriority::ConflictEarly(_)) {
+                    // Already in the right category
+                    return false;
+                };
+                let index = Self::get_index(&entry).unwrap_or(next);
+                entry.insert(PubGrubPriority::ConflictEarly(Reverse(index)));
+                true
+            }
+            std::collections::hash_map::Entry::Vacant(entry) => {
+                entry.insert(PubGrubPriority::ConflictEarly(Reverse(next)));
+                true
+            }
+        }
+    }
+
+    pub(crate) fn make_conflict_late(&mut self, package: &PubGrubPackage) -> bool {
```

Comment:
Can you add a comment here, and clarify how it's different from `make_conflict_early`?